ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,129 @@
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+
+        std::cout << "Capturing a frame" << std::endl;
+        const auto frame = camera.capture();
+
+        const auto intrinsics = camera.intrinsics();
+
+        std::cout << "Setting up visualization" << std::endl;
+        Zivid::CloudVisualizer vis;
+        zivid.setDefaultComputeDevice(vis.computeDevice());
+
+        std::cout << "Displaying the point cloud" << std::endl;
+        vis.showMaximized();
+        vis.show(frame);
+        vis.resetToFit();
+
+        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+        vis.run();
+
+        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+
+        const auto pointCloud = frame.getPointCloud();
+        cv::Mat rgb((int)pointCloud.height(), (int)pointCloud.width(), CV_8UC3, cv::Scalar(0, 0, 0));
+        pointCloudToRGB(pointCloud, rgb);
+
+        std::cout << "Undistorting the RGB image" << std::endl;
+
+        cv::Size size = rgb.size();
+
+        cv::Mat rgbUndistorted;
+        cv::Mat rgbUndistortedFull;
+
+        cv::Mat distortionCoefficients(cv::Size(1, 5), CV_64FC1, cv::Scalar(0));
+        cv::Mat cameraMatrix(cv::Size(3, 3), CV_64FC1, cv::Scalar(0));
+        cv::Mat optimalCameraMatrix;
+
+        getCameraIntrinsics(camera, distortionCoefficients, cameraMatrix);
+
+        optimalCameraMatrix = cv::getOptimalNewCameraMatrix(cameraMatrix, distortionCoefficients, size, 1, size);
+
+        cv::undistort(rgb, rgbUndistorted, cameraMatrix, distortionCoefficients);
+        cv::undistort(rgb, rgbUndistortedFull, cameraMatrix, distortionCoefficients, optimalCameraMatrix);
+
+        std::cout << "Displaying and saving the RGB image" << std::endl;
+
+        displayRGB(rgb, "Distorted RGB image");
+        cv::imwrite("Distorted RGB image.jpg", rgb);
+
+        std::cout << "Displaying and saving the undistorted RGB image" << std::endl;
+
+        displayRGB(rgbUndistorted, "Undistorted RGB image");
+        cv::imwrite("Undistorted RGB image.jpg", rgbUndistorted);
+
+        std::cout << "Displaying and saving the Undistorted RGB image - full" << std::endl;
+
+        displayRGB(rgbUndistortedFull, "Undistorted RGB image - full");
+        cv::imwrite("Undistorted RGB image - full.jpg", rgbUndistorted);
+    }
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+void pointCloudToRGB(const Zivid::PointCloud &pointCloud, cv::Mat &rgb)
```

Comment:
We should avoid out parameters like `cv::Mat &rgb`, and instead use return values. Move the initialization of `rgb` inside the `pointCloudToRGB` function, and then return it by value. (The resulting `rgb` variable in `main()` should then also be made `const`)
The C++ Core Guidelines have a [section about this](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f20-for-out-output-values-prefer-return-values-to-output-parameters), although only the first line is really relevant in this case.

Reply:
Done. I think.